report/validation/thermodesorption_spectra/dark/dark_tds.md

```diff
@@ -0,0 +1,256 @@
+T_imp = 370 # K
+T_rest = 295 # K
+R_p = 0.7e-9 # m
+sigma = 0.5e-9 # m
+t_imp = 72 * 3600 # s
+t_rest = 12 * 3600 # s
+Beta = 0.05 # K s^-1
+
+fluence = 1.5e25
+flux = fluence / t_imp
+
+# Deuterium Beam Profile (S = flux * f(x))
+distribution = (1 / (sigma * (2 * np.pi) ** 0.5) * sp.exp(-0.5 * ((F.x - R_p) / sigma) ** 2))
+
+model = F.Simulation()
+
+vertices = np.concatenate(
+    [
+        np.linspace(0, 3e-9, num=100),
+        np.linspace(3e-9, 8e-6, num=100),
+        np.linspace(8e-6, 8e-5, num=100),
+        np.linspace(8e-5, 8e-4, num=100),
+    ]
+)
+model.mesh = F.MeshFromVertices(vertices)
+
+# material
+damaged_tungsten = F.Material(1, D_0, E_D, borders=[0, sample_depth])
+model.materials = damaged_tungsten
+
+# hydrogen source
+ion_flux = sp.Piecewise((flux*distribution, F.t < t_imp), (0, True))
+source_term = F.Source(value=ion_flux, volume=1, field=0)
+model.sources = [source_term]
+
+# boundary conditions
+model.boundary_conditions = [F.DirichletBC(surfaces=[1, 2], value=0, field="solute")]
+
+# temperature
+start_tds = t_imp + t_rest  # s
+min_temp, max_temp = 300, 1000
+
+model.T = F.Temperature(
+    value=sp.Piecewise(
+        (T_imp, F.t < t_imp),
+        (T_rest, F.t < start_tds),
+        (min_temp + Beta * (F.t - start_tds), True),
+    )
+)
+
+# trap settings
+k_0 = D_0 / (1.1e-10**2 * 6 * w_atom_density)
+damage_dist = 1 / (1 + sp.exp((F.x - 2.5e-06) / 5e-07))
+n_i = np.array([2.4e-3, 4.8, 3.8, 2.6, 3.6, 1.1]) * 1e25
+neutron_induced_traps = []
+E_p = [1.04, 1.15, 1.35, 1.65, 1.85, 2.05]
+
+trap_1 = F.Trap(
+    E_k = 0.28,
+    k_0 = k_0,
+    E_p = 1.04,
+    p_0 = 1e13,
+    density = n_i[0],
+    materials = damaged_tungsten
+)
+
+for i in range(1, 6):
+    neutron_induced_traps.append(F.Trap(
+        k_0 = k_0,
+        E_k = E_D,
+        p_0 = 1e13,
+        E_p = E_p[i],
+        density = n_i[i] * damage_dist,
+        materials = damaged_tungsten,
+    ))
+
+traps = [trap_1] + neutron_induced_traps
+model.traps = traps
+
+model.dt = F.Stepsize(
+    initial_value=1,
+    stepsize_change_ratio=1.1,
+    t_stop=t_imp + t_rest * 0.5,
+    dt_min=1e-1,
+    stepsize_stop_max=50,
+)
+
+model.settings = F.Settings(
+    absolute_tolerance=1e10,
+    relative_tolerance=1e-10,
+    final_time=start_tds + (max_temp - min_temp) / Beta,  # time to reach max temp
+)
+
+derived_quantities = F.DerivedQuantities(
+    [
+        F.AverageVolume("T", volume=1),
+        F.TotalVolume("solute", volume=1),
+        F.TotalVolume("retention", volume=1),
+        F.TotalVolume("1", volume=1),
+        F.TotalVolume("2", volume=1),
+        F.TotalVolume("3", volume=1),
+        F.TotalVolume("4", volume=1),
+        F.TotalVolume("5", volume=1),
+        F.TotalVolume("6", volume=1),
+        F.HydrogenFlux(surface=1),
+        F.HydrogenFlux(surface=2),
+    ],
+)
+
+model.exports = [derived_quantities]
+
+model.initialise()
+model.run()
+```
+
+```{note}
+Implementation details from {cite}`dark_modelling_2024_code`.
```

Comment:
In reality the implementation details are available in the paper, this will be a copy of the initial implementation. Since you will add the other doses, I would rephrase this as "Experimental data taken from [...]"